src/index.ts

```diff
@@ -149,32 +149,19 @@ class ClientApp {
         pcOrigin?: string;
     } = {}) {
         if (cfg) {
-            if (cfg.hasOwnProperty('pcEnvironmentQueryParam')) {
+            if ('pcEnvironmentQueryParam' in cfg) {
                 const paramName = cfg.pcEnvironmentQueryParam;
-
                 if (typeof paramName !== 'string' || paramName.trim().length === 0) {
                     throw new Error('Invalid query param name provided.  Must be non-null, non-empty string');
                 }
-
                 const parsedQueryString = queryString.parse(ClientApp._getQueryString() || '');
-                const paramValue = parsedQueryString[paramName];
-                if (paramValue && typeof paramValue === 'string') {
-                    this._pcEnv = envUtils.lookupPcEnv(paramValue, true);
-                    if (!this._pcEnv) {
-                        throw new Error(`Could not parse '${paramValue}' into a known PureCloud environment`);
-                    }
-                } else {
-                    throw new Error(`Could not find unique value for ${paramName} parameter on Query String`);
-                }
-            } else if (cfg.hasOwnProperty('pcEnvironment')) {
+                this._pcEnv = this.lookupEnv(parsedQueryString[paramName]);
```

Comment:
The queryString lib can actually return a string or a string[] depending if their are duplicate keys.

Looks like envUtils.lookupPcEnv will fail out if the type passed is not string, but seems a bit brittle to rely on this.  Should we attempt to narrow the type here and fail if it's not a string, or, were you ok with the pass through and fail from envUtils?